services/management/dbaas/components_service.go

```diff
@@ -392,3 +390,90 @@ func setComponent(kc *models.Component, rc *dbaasv1beta1.ChangeComponent) (*mode
 	kc.DisabledVersions = stringset.ToSlice(disabledVersions)
 	return kc, nil
 }
+
+func (c componentsService) unsetDefaultVersionIfUnsupported(ctx context.Context, operatorType, operatorVersion string, component *models.Component) (bool, error) {
+	if component == nil {
+		return false, nil
+	}
+	supported, err := c.versionServiceClient.IsDatabaseVersionSupportedByOperator(ctx, operatorType, operatorVersion, component.DefaultVersion)
+	if !supported {
+		if err != nil {
+			return false, err
+		}
+		// When default version is empty, the latest recommended version is used.
+		component.DefaultVersion = ""
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c componentsService) InstallOperator(ctx context.Context, req *dbaasv1beta1.InstallOperatorRequest) (*dbaasv1beta1.InstallOperatorResponse, error) {
+	kubernetesCluster, err := models.FindKubernetesClusterByName(c.db.Querier, req.KubernetesClusterName)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	parsedPMMVersion, err := goversion.NewVersion(pmmversion.PMMVersion)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	pmmVersion := parsedPMMVersion.Core().String()
+
+	// Just check the version we are asked to update to is present in version service.
+	resp, err := c.versionServiceClient.Matrix(ctx, componentsParams{product: "pmm-server", productVersion: pmmVersion})
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	if len(resp.Versions) == 0 {
+		return nil, errors.Errorf("failed to validate operator version")
```

Comment:
```suggestion
		return nil, errors.Errorf("failed to validate PMM version")
```

Reply:
failed to get operators compatible with PMM version